accessgraph sync: Add AWS IAM role for EKS audit logs

camscale opened the pull request:
Update the `teleport configure integration acces-graph aws-iam` command
to add a permission to access EKS audit logs via CloudWatch Logs if the
`--eks-audit-logs` flag is passed. This is necessary so that an
integration can pull the EKS audit logs if so configured in a discovery
access graph matcher.

Issue: https://github.com/gravitational/access-graph/issues/1589

camscale committed:
accessgraph sync: Add AWS IAM role for EKS audit logs

Update the `teleport configure integration acces-graph aws-iam` command
to add a permission to access EKS audit logs via CloudWatch Logs if the
`--eks-audit-logs` flag is passed. This is necessary so that an
integration can pull the EKS audit logs if so configured in a discovery
access graph matcher.

marcoandredinis reviewed (approved):
I would also add the new param here
https://github.com/gravitational/teleport/blob/57a7f8caa686699a940d55ae73e6840da3f7b764/lib/web/integrations_awsoidc.go#L1339

I think it's the only required change in the backend that is missing.

camscale committed:
web: Add web eksAuditLogs to integration configure endpoint

Extend the web endpoint for the webscript for integrations configure
access-graph-cloud-sync-iam.sh to add the `eksAuditLogs` query param to
configure with EKS audit logs enabled. Add tests for this endpoint as
there were none.
